src/main/java/org/apache/commons/compress/archivers/ar/ArArchiveInputStream.java

```diff
@@ -241,102 +232,172 @@ private String getExtendedName(final int offset) throws IOException {
+            final byte[] headerBuf = getRecord();
+            if (headerBuf == null) {
+                // If we encountered special records but no file entry, the archive is malformed
+                if (foundGNUStringTable) {
+                    throw new EOFException("Premature end of ar archive: no regular entry after GNU string table.");
+                }
+                currentEntry = null;
+                return null; // End of archive
             }
-            trackReadBytes(1);
-        }
-        {
-            final int read = IOUtils.readFully(in, metaData);
-            trackReadBytes(read);
-            if (read == 0) {
-                return null;
+            checkTrailer();
+
+            // Parse the header into a new entry
+            currentEntry = parseEntry(headerBuf);
+            entryOffset = getBytesRead(); // Store the offset of the entry
+
+            foundGNUStringTable = isGNUStringTable(currentEntry);
+            if (foundGNUStringTable) {
+                // If this is a GNU string table entry, read the extended names and continue
+                namebuffer = readGNUStringTable(currentEntry);
             }
-            if (read < metaData.length) {
-                throw new ArchiveException("Truncated ar archive");
+        } while (foundGNUStringTable);
+
+        // Handle long file names and other special cases
+        String name = currentEntry.getName();
+        long len = currentEntry.getLength();
+        // Handle GNU ar: names ending with '/' are terminated (allows spaces in names)
+        if (name.endsWith("/")) {
+            name = name.substring(0, name.length() - 1);
+        } else if (isGNULongName(name)) {
+            // GNU ar: name is a reference to the string table (e.g., "/42"), resolve the actual name
+            final int off = ParsingUtils.parseIntValue(name.substring(1));
+            name = getExtendedName(off);
+        } else if (isBSDLongName(name)) {
+            // BSD ar: name is stored after the header, retrieve it
+            name = getBSDLongName(name);
+            // The entry length includes the file name length; adjust to get the actual file data length
+            final int nameLen = name.length();
+            if (nameLen > len) {
+                throw new ArchiveException(
+                        "Invalid BSD long name: file name length (" + nameLen + ") exceeds entry length (" + len + ")");
             }
+            len -= nameLen;
+            entryOffset += nameLen;
         }
-        {
-            final byte[] expected = ArchiveUtils.toAsciiBytes(ArArchiveEntry.TRAILER);
-            final byte[] realized = IOUtils.readRange(in, expected.length);
-            final int read = realized.length;
-            trackReadBytes(read);
-            if (read != expected.length) {
-                throw new ArchiveException("Failed to read entry trailer. Occurred at byte: " + getBytesRead());
+
+        currentEntry = new ArArchiveEntry(
```

Comment:
I'm not a fan of what I call "newspaper" formatting. Why is this API call in a column and not another?